haystack/nodes/file_converter/parsr.py

```diff
@@ -243,6 +243,10 @@ def _convert_text_element(self, element: Dict[str, Any]) -> str:
                 current_paragraph = "\n".join([self._get_paragraph_string(elem) for elem in element["content"]])
                 return current_paragraph
 
+        if element["type"] == "list":
```

Comment:
Should we make this an `elif` statement since they are mutually exclusive?

Also since the two lines above are exactly the same, I was thinking maybe we can do something like this?

```
        if element["type"] in ["table-of-contents", "list"]:
            if self.remove_table_of_contents:
                return ""
            current_paragraph = "\n".join([self._get_paragraph_string(elem) for elem in element["content"]])
            return current_paragraph
```

Reply:
IMHO, no need for `elif` as all if blocks contain a return statement. And I agree, we can merge the `table-of-contents` and `list` case - would need to look like something like this then to not remove lists if `remove_table_of_contents` is set to `True`:
```python
if element["type"] in ["table-of-contents", "list"]:
    if self.remove_table_of_contents and element["type"] == "table-of-contents":
        return ""
    current_paragraph = "\n".join([self._get_paragraph_string(elem) for elem in element["content"]])
    return current_paragraph
```

Reply:
Last suggestion applied